activity.go

```diff
@@ -56,8 +61,14 @@ type (
 // Serialization of all primitive types, structures is supported ... except channels, functions, variadic, unsafe pointer.
 // This method calls panic if activityFunc doesn't comply with the expected format.
 func RegisterActivity(activityFunc interface{}) {
+	RegisterActivityWithOptions(activityFunc, RegisterActivityOptions{})
+}
+
+// RegisterActivityWithOptions registers an activity function with the framework with provided
+// options.
+func RegisterActivityWithOptions(activityFunc interface{}, options RegisterActivityOptions) {
```

Comment:
Should this method that allows us to register with a name be on the Worker instead of global? Wasn't the concern with "custom names" that there could much easier be conflicts if we allow global registration?